controller/lib/hal/hal.h

```diff
@@ -236,6 +236,9 @@ class HalApi {
   // Performs the device soft-reset
   [[noreturn]] void reset_device();
 
+  // Start the high priority loop timer
+  void startLoopTimer(const Duration &period, void (*callback)());
```

Comment:
Is it possible to use an std::function<void()> instead of a function pointer? That way it can capture variables so the caller isn't obligated to use static variables.

Reply:
If not, then please at least add a "void* data" argument to the callback and to startLoopTimer. And document what are the lifetime requirements on the data argument (basically it must stay alive until the callback is reset or until the device shuts down, right?)

Reply:
I can add an argument to the function although it's not currently being used.
I'm not really envisioning us changing the high priority function that is used, my intent was to call startLoopTimer just once at startup and never change the callback function.
I suppose we could add provisions for changing the callback function if we thought it would make the code clearer, like have a different callback function for different modes of the system.  That might be too cleaver by half though.
As to std:function(), I'm not really familiar with the syntax and a brief Google search only served to scare me off.  I'm certainly not opposed to someone changing this to a more modern style, but for now I think I'll just stick with a good old function pointer which is what I'm familiar with.